### PR TITLE
Docfix warnings

### PR DESCRIPTION
**Pull request by mfkl**
Makes progress towards https://github.com/Microsoft/msbuild/issues/2927

### CODE REVIEW

File: src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <IsPackable>true</IsPackable>
     <PackageDescription>This package contains the $(MSBuildProjectName) assembly which contains logic for converting projects.  NOTE: This assembly is deprecated.</PackageDescription>
     <IncludeSatelliteOutputInPack>false</IncludeSatelliteOutputInPack>
+    <NoWarn>$(NoWarn);1570;1572;1573;1587</NoWarn>
```
Comment (Forgind) [inline src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:10]: If all the docs are properly formatted, now, you should be able to delete this line, right?

File: src/Deprecated/Engine/Microsoft.Build.Engine.csproj [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <IsPackable>true</IsPackable>
     <PackageDescription>This package contains the $(MSBuildProjectName) assembly which contains the legacy compatibility shim for the MSBuild engine.  NOTE: This assembly is deprecated.</PackageDescription>
     <IncludeSatelliteOutputInPack>false</IncludeSatelliteOutputInPack>
+    <NoWarn>$(NoWarn);1570;1572;1573;1587</NoWarn>
```
Comment (Forgind) [inline src/Deprecated/Engine/Microsoft.Build.Engine.csproj:16]: Same goes for this line.

File: src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj (hunk 1 above)
Comment (mfkl) [reply src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:10]: As mentioned in https://github.com/dotnet/msbuild/pull/5575/commits/fe93fcadd754ad0789aa6614ba4ae2ac19f27d6b, the deprecated projects had another couple hundreds errors to fix so I decided to make a PR first for the non-deprecated code.

Comment (Forgind) [reply src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:10]: That sounds reasonable to me, but I'd hesitate to close the issue until those are resolved, too.

Comment (mfkl) [reply src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:10]: you're right; removed "Fixes" from the last commit so GH doesn't close the issue.

**Comment by elachlan** [comment]
@mfkl there are merge conflicts.

**Comment by Forgind** [comment]
I resolved them

**Comment by Forgind** [comment]
Thanks, @mfkl!

**Comment by elachlan** [comment]
@ghogen shouldn't we do another PR for those changes since this is merged?

**Comment by ghogen** [comment]
@elachlan Yes, do you want to do it.  I haven't contributed yet to this repo, but I could probably also submit a PR for these.

**Pull request merged**